src/prefect/filesystems.py

```diff
@@ -418,15 +457,18 @@ async def get_directory(
         )
 
     async def put_directory(
```

Comment:
_Almost_ - the subclasses of `RemoteFileSystem` have `put_directory` calls that call `self.filesystem.put_directory`; I initially tried to implement them as subclasses of `RemoteFileSystem` so that we wouldn't need to duplicate the signature but it turns out subclassing Blocks was non-trivial.